2020-02-08  Cindi Dong  <cindi@lnxsrv09.seas.ucla.edu>

        * timespec.h (timespec_cmp): Updated to use comparison rather than 
subtraction, which may result in overflow.
--- /u/be/ugrad/cindi/coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec.h	2020-02-08 22:03:04.039085000 -0800
@@ -45,10 +45,29 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec > b.tv_sec)
+    {
+      return 1;
+    }
+  else if (a.tv_sec < b.tv_sec)
+    {
+      return -1;
+    }
+  else {
+    if (a.tv_nsec > b.tv_nsec)
+      {
+	return 1;
+      }
+    else if (a.tv_nsec < b.tv_nsec)
+      {
+	return -1;
+      }
+    else
+      {
+	return 0;
+      }
+  }
 }
-
 # if ! HAVE_DECL_NANOSLEEP
 /* Don't specify a prototype here.  Some systems (e.g., OSF) declare
    nanosleep with a conflicting one (const-less first parameter).  */
